internal/app/skuba/node/join.go

```diff
@@ -41,8 +42,15 @@ func NewJoinCmd() *cobra.Command {
 		Use:   "join <node-name>",
 		Short: "Joins a new node to the cluster",
 		Run: func(cmd *cobra.Command, nodenames []string) {
+			if err := validate.NodeName(nodenames[0]); err != nil {
+				klog.Fatal(err)
+			}
+
 			joinConfiguration := deployments.JoinConfiguration{
-				KubeadmExtraArgs: map[string]string{"ignore-preflight-errors": joinOptions.ignorePreflightErrors},
+				KubeadmExtraArgs: map[string]string{
+					"ignore-preflight-errors": joinOptions.ignorePreflightErrors,
+					"node-name":               nodenames[0],
```

Comment:
This is already performed here: https://github.com/SUSE/skuba/blob/e0de7024acc781dc2086a51dfeab01b033f85e82/pkg/skuba/actions/node/join/join.go#L136-L138 as well.